In [94]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

## Data Injestion

In [95]:
possum = pd.read_csv('possum.csv')
possum

case  site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0       1     1    Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1       2     1    Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2       3     1    Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3       4     1    Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4       5     1    Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   
..    ...   ...    ...  ..  ...      ...     ...       ...    ...       ...   
99    100     7  other   m  1.0     89.5    56.0      81.5   36.5      66.0   
100   101     7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
101   102     7  other   f  6.0     92.4    55.0      89.0   38.0      63.5   
102   103     7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   
103   104     7  other   f  3.0     93.6    59.9      89.0   40.0      67.6   

     earconch   eye  chest  belly  
0        54.5  15.2   28.0   36.0  
1        51.2  16.0   28.5   33.0  
2        51.9  15.5   30.0   34.0  
3        52.2  15.2   28.0   34.0  
4        53.2  15.1   28.5   33.0  
..        ...   ...    ...    ...  
99       46.8  14.8   23.0   27.0  
100      48.0  14.0   25.0   33.0  
101      45.4  13.0   25.0   30.0  
102      45.9  15.4   25.0   29.0  
103      46.0  14.8   28.5   33.5  

[104 rows x 14 columns]

## Data Cleaning

In [96]:
possum.dropna(inplace=True)

In [116]:
possum.duplicated().sum()

np.int64(0)

## Preliminary Data Analysis 

In [97]:
possum.head()

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3     4     1  Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4     5     1  Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0  
3      52.2  15.2   28.0   34.0  
4      53.2  15.1   28.5   33.0

In [98]:
possum.tail()

case  site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
99    100     7  other   m  1.0     89.5    56.0      81.5   36.5      66.0   
100   101     7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
101   102     7  other   f  6.0     92.4    55.0      89.0   38.0      63.5   
102   103     7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   
103   104     7  other   f  3.0     93.6    59.9      89.0   40.0      67.6   

     earconch   eye  chest  belly  
99       46.8  14.8   23.0   27.0  
100      48.0  14.0   25.0   33.0  
101      45.4  13.0   25.0   30.0  
102      45.9  15.4   25.0   29.0  
103      46.0  14.8   28.5   33.5

In [ ]:
possum_copy = possum.copy()
possum.set_index("case", inplace=True)

In [117]:
possum.info()

<class 'pandas.core.frame.DataFrame'>
Index: 101 entries, 0 to 103
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   case      101 non-null    int64  
 1   site      101 non-null    int64  
 2   Pop       101 non-null    object 
 3   sex       101 non-null    object 
 4   age       101 non-null    float64
 5   hdlngth   101 non-null    float64
 6   skullw    101 non-null    float64
 7   totlngth  101 non-null    float64
 8   taill     101 non-null    float64
 9   footlgth  101 non-null    float64
 10  earconch  101 non-null    float64
 11  eye       101 non-null    float64
 12  chest     101 non-null    float64
 13  belly     101 non-null    float64
dtypes: float64(10), int64(2), object(2)
memory usage: 11.8+ KB


## Descriptive Analysis

In [118]:
possum.describe()
possum_des = possum.describe()

In [ ]:
#Calculating the mode for possums
possum_des.loc['mode', 'case':'belly'] = [stats.mode(possum[feature])[0] for feature in male_des.columns]

# Calculating the variance of possums
possum_des.loc['var', 'case': 'belly'] = possum_des.loc['std', 'case': 'belly'] ** 2

#Calculating the  kurtosis of possums
possum_des.loc['kurtosis'] = [stats.kurtosis(possum[feature]) for feature in possum_des.columns]

#Calculating the  IQR of possums
possum_des.loc['IQR'] = possum_des.loc['75%', 'case':'belly'] -  possum_des.loc['25%', 'case':'belly']

#Calculating the  SKEWNESS of possums
possum_des.loc['Skewness'] = [stats.skew(possum[feature]) for feature in possum_des.columns]
possum_des

case        site         age     hdlngth      skullw  \
count     101.000000  101.000000  101.000000  101.000000  101.000000   
mean       52.762376    3.673267    3.821782   92.730693   56.960396   
std        30.573894    2.366892    1.915182    3.518714    3.102679   
min         1.000000    1.000000    1.000000   82.500000   50.000000   
25%        26.000000    1.000000    2.000000   90.700000   55.000000   
50%        54.000000    4.000000    3.000000   92.900000   56.400000   
75%        79.000000    6.000000    5.000000   94.800000   58.100000   
max       104.000000    7.000000    9.000000  103.100000   68.600000   
var       934.762970    5.602178    3.667921   12.381349    9.626616   
kurtosis   -1.242504   -1.586846   -0.317807    0.925124    2.337963   
IQR        53.000000    5.000000    3.000000    4.100000    3.100000   
Skewness   -0.025027    0.134643    0.547475   -0.054117    1.016620   

            totlngth       taill    footlgth    earconch         eye  \
count     101.000000  101.000000  101.000000  101.000000  101.000000   
mean       87.269307   37.049505   68.398020   48.133663   15.050495   
std         4.196802    1.971681    4.413502    4.060352    1.058643   
min        75.000000   32.000000   60.300000   41.300000   12.800000   
25%        84.500000   36.000000   64.500000   44.800000   14.400000   
50%        88.000000   37.000000   67.900000   46.800000   14.900000   
75%        90.000000   38.000000   72.500000   52.000000   15.700000   
max        96.500000   43.000000   77.900000   56.200000   17.800000   
var        17.613149    3.887525   19.478996   16.486455    1.120725   
kurtosis   -0.212877    0.311398   -1.195542   -1.379705   -0.023105   
IQR         5.500000    2.000000    8.000000    7.200000    1.300000   
Skewness   -0.245144    0.121612    0.122903    0.215128    0.376387   

               chest       belly  
count     101.000000  101.000000  
mean       27.064356   32.638614  
std         2.020722    2.727745  
min        22.000000   25.000000  
25%        25.500000   31.000000  
50%        27.000000   32.500000  
75%        28.000000   34.000000  
max        32.000000   40.000000  
var         4.083317    7.440594  
kurtosis   -0.282777    0.167455  
IQR         2.500000    3.000000  
Skewness   -0.057535    0.104267

``Analysis on Female possum``

In [115]:
female_possum = possum[possum['sex'] == 'f']


In [114]:
female_des = female_possum.describe()


In [129]:
#Calculating the mode of the male specie
female_des.loc['mode', 'case':'belly'] = [stats.mode(female_possum[feature])[0] for feature in female_des.columns]

# Calculating the variance of the female specie
female_des.loc['var', 'case': 'belly'] = female_des.loc['std', 'case': 'belly'] ** 2

#Calculating the  kurtosis of the female specie
female_des.loc['kurtosis'] = [stats.kurtosis(female_possum[feature]) for feature in female_des.columns]

#Calculating the  IQR of the female specie
female_des.loc['IQR'] = female_des.loc['75%', 'case':'belly'] -  female_des.loc['25%', 'case':'belly']

#Calculating the  SKEWNESS of the female specie
female_des.loc['Skewness'] = [stats.skew(female_possum[feature]) for feature in female_des.columns]
female_des

case       site        age    hdlngth     skullw   totlngth  \
count      42.000000  42.000000  42.000000  42.000000  42.000000  42.000000   
mean       43.476190   3.000000   3.952381  92.238095  56.578571  88.023810   
std        30.629364   2.241515   1.962482   2.537903   2.599112   4.161309   
min         2.000000   1.000000   1.000000  84.700000  51.500000  75.000000   
25%        17.500000   1.000000   3.000000  91.075000  55.100000  85.625000   
50%        39.500000   2.000000   3.500000  92.500000  56.350000  88.750000   
75%        64.750000   5.000000   5.000000  93.900000  57.675000  90.500000   
max       104.000000   7.000000   9.000000  96.900000  67.700000  96.500000   
mode        2.000000   1.000000   3.000000  90.000000  56.300000  89.000000   
var       938.157956   5.024390   3.851336   6.440952   6.755383  17.316492   
kurtosis   -1.034701  -1.264021   0.120178   0.882062   6.388643   0.983858   
IQR        47.250000   4.000000   2.000000   2.825000   2.575000   4.875000   
Skewness    0.344977   0.565514   0.694254  -0.828588   1.652190  -0.635726   

              taill   footlgth   earconch        eye      chest      belly  
count     42.000000  42.000000  42.000000  42.000000  42.000000  42.000000  
mean      37.119048  69.111905  48.773810  14.785714  27.345238  32.940476  
std        1.850542   4.911321   4.123819   1.028276   1.862624   2.940886  
min       32.000000  60.300000  41.300000  13.000000  23.000000  25.000000  
25%       36.000000  64.850000  45.150000  14.050000  26.000000  31.500000  
50%       37.750000  70.450000  50.800000  14.750000  28.000000  33.000000  
75%       38.375000  72.800000  52.350000  15.400000  28.500000  34.000000  
max       41.000000  77.900000  53.900000  17.400000  31.000000  40.000000  
mode      38.000000  72.800000  46.000000  14.000000  28.000000  34.000000  
var        3.424506  24.121074  17.005883   1.057352   3.469367   8.648810  
kurtosis   0.117205  -1.156776  -1.494006   0.006988  -0.441086   0.822925  
IQR        2.375000   7.950000   7.200000   1.350000   2.500000   2.500000  
Skewness  -0.477052  -0.138333  -0.280441   0.361100  -0.128651   0.119819

``Analysis on Male Possum``

In [112]:
male_possum = possum[possum['sex'] == 'm']


In [113]:
male_des = male_possum.describe()


In [128]:
#Calculating the mode of the male specie
male_des.loc['mode', 'case':'belly'] = [stats.mode(male_possum[feature])[0] for feature in male_des.columns]

#Calculating the  Kurtosis of the male specie
male_des.loc['kurtosis'] = [stats.kurtosis(male_possum[feature]) for feature in male_des.columns]

#Calculating the  VARIANCE of the male specie
male_des.loc['var', 'case': 'belly'] = male_des.loc['std', 'case': 'belly'] ** 2

#Calculating the  IQR of the male specie
male_des.loc['IQR'] = male_des.loc['75%', 'case':'belly'] -  male_des.loc['25%', 'case':'belly']

#Calculating the  Skewness of the male specie
male_des.loc['Skewness'] = [stats.skew(male_possum[feature]) for feature in male_des.columns]
male_des

case       site        age     hdlngth     skullw   totlngth  \
count      59.000000  59.000000  59.000000   59.000000  59.000000  59.000000   
mean       59.372881   4.152542   3.728814   93.081356  57.232203  86.732203   
std        29.010045   2.354845   1.892185    4.061190   3.411969   4.173828   
min         1.000000   1.000000   1.000000   82.500000  50.000000  77.000000   
25%        34.500000   2.000000   2.000000   90.700000  55.050000  84.000000   
50%        61.000000   5.000000   3.000000   93.300000  56.600000  86.000000   
75%        84.500000   6.000000   5.000000   95.400000  59.000000  89.750000   
max       103.000000   7.000000   8.000000  103.100000  68.600000  96.000000   
var       841.582700   5.545295   3.580362   16.493267  11.641531  17.420842   
mode        1.000000   1.000000   3.000000   89.500000  56.000000  85.000000   
kurtosis   -1.169269  -1.547001  -0.750377    0.312342   1.054077  -0.726422   
IQR        50.000000   4.000000   3.000000    4.700000   3.950000   5.750000   
Skewness   -0.251137  -0.160326   0.426401   -0.077762   0.706645   0.015491   

              taill   footlgth   earconch        eye      chest      belly  
count     59.000000  59.000000  59.000000  59.000000  59.000000  59.000000  
mean      37.000000  67.889831  47.677966  15.238983  26.864407  32.423729  
std        2.067816   3.987683   3.986921   1.047825   2.118943   2.569405  
min       32.000000  62.000000  41.700000  12.800000  22.000000  27.000000  
25%       35.750000  64.500000  44.800000  14.500000  25.500000  30.750000  
50%       36.500000  66.500000  46.200000  15.000000  27.000000  32.000000  
75%       38.000000  71.500000  51.700000  15.950000  28.000000  35.000000  
max       43.000000  75.000000  56.200000  17.800000  32.000000  38.000000  
var        4.275862  15.901619  15.895541   1.097937   4.489918   6.601841  
mode      38.000000  65.600000  44.900000  15.000000  27.000000  32.000000  
kurtosis   0.417221  -1.313311  -0.969600  -0.093204  -0.234716  -0.795638  
IQR        2.250000   7.000000   6.900000   1.450000   2.500000   4.250000  
Skewness   0.438093   0.273069   0.588725   0.413286   0.040812   0.012574

``General Analysis on possums``
1. We have 101 observation
2. The average age is 3.82 and the age at the center of the dataset is 3 and most of the observation have the age of 3
3. The average size is 87.3, the median is 88 and most observation have the size of 89.
4. Most features are normal distribution since their mean, median, mode are nearly equal.


``Analysis on female possums``


``Comparison on the Male and Female Specie``

1. The percentage difference between the number males and female is: 16% i.e there were more male than female
2. With regards to age, the maximum age of female is 9 while that of male is 8
3. The female are bigger than the male, though the male have bigger heads than the female
4. The male have bigger eyes and skull.
5. The female have bigger belly and chest than the male.


In [105]:
# print('====Analysis on Case======')


# mean_value = male_possum.iloc[].mean()
# print(F'Mean value:  {round(mean_value)}')

# median_value = male_possum['case'].median()
# print(F'Median value:  {round(median_value)}')

# variance_value = male_possum['case'].var()   
# print(f'Variance value : {round(variance_value)}')

# std_dev_value = male_possum['case'].std(ddof=0)    
# print(f'Standard deviation: {round(std_dev_value)}')

# scores = male_possum['site'].values
# q1 = np.quantile(scores, 0.25)
# q2 = np.quantile(scores, 0.50)  # median
# q3 = np.quantile(scores, 0.75)

# print(f"Q1 - {q1}")
# print(f"Q2 - {q2}")
# print(f"Q3 - {q3}")

# # Interquartile Range
# iqr = q3 - q1
# print(f"IQR - {iqr}")

# # Calculate skewness
# skewness = male_possum['case'].skew()

# print(f"Skewness: {skewness:.3f}")

# kurtosis = male_possum['case'].kurtosis()

# print(f"kurtosis: {kurtosis:.3f}")


